### ISSUE TITLE
Editor can't save dashboard when json has been changed

### ISSUE THREAD

**Issue by vasili439** [grafana/grafana#23609]
**What happened**:
Editor can't save dashboard when json has been changed

**What you expected to happen**:
Dashboard saved

**How to reproduce it (as minimally and precisely as possible)**:
1. Add user with Viewer permissions
2. Create team Team1 and add user to it
3. Set Editor permissions for folder Folder1 for Team1
4. Save as any dashboard from Folder1 to Folder1 with new name
5. Try to edit new dashboard's json and click Save
6. You will get "Access denied to save dashboard"

grafana logs:
```
t=2020-04-16T11:28:50+0300 lvl=eror msg="Access denied to save dashboard" logger=context userId=62 orgId=1 uname=test_test error="Access denied to save dashboard" remote_addr=1.1.1.1
t=2020-04-16T11:28:50+0300 lvl=info msg="Request Completed" logger=context userId=62 orgId=1 uname=test_test method=POST path=/api/dashboards/db status=403 remote_addr=1.1.1.1 time_ms=23 size=73 referer="https://grafana.dev.domain.net/d/YyDuQZqZz/dashboard-copy-123?editview=dashboard_json&orgId=1"
```

**Anything else we need to know?**:
grafana server is behind nginx

**Environment**:
- Grafana version: 6.7.2
- Data source type & version:
- OS Grafana is installed on:  CentOS 8
- User OS & Browser: Win/Mac, Chrome
- Grafana plugins: alexanderzobnin-zabbix-app  btplc-trend-box-panel  grafana-image-renderer  grafana-piechart-panel  grafana-worldmap-panel
- Others:

**Comment by hugohaggmark** [comment]
@vasili439 thank you for reporting this issue!

**Comment by vasili439** [comment]
Important notes: we are using LDAP. Users have default privileges: Viewer but if I add him to Editor group everything is works as expected. So conclusion could be: grafana cannot escalate user's privileges if they assigned to particular folder

**Comment by marefr** [comment]
Think this have been fixed by #23314. Please try 6.7.3 or Grafana latest stable
